src/cmd/services/m3coordinator/downsample/options.go

```diff
@@ -753,6 +753,9 @@ func (cfg Configuration) newAggregator(o DownsamplerOptions) (agg, error) {
 			zap.Int("mapping_rules", len(cfg.Rules.MappingRules)),
 			zap.Int("rollup_rules", len(cfg.Rules.RollupRules)),
 		)
+		// Emit number of mapping rules and rollup rules received from configs
+		scope.Gauge("mapping_rules").Update(float64(len(cfg.Rules.MappingRules)))
```

Comment:
Can you add a piece of comment for the complete metric name? It's always frustrating that code search a m3 metric name doesn't surface the source code piece. At Facebook, one coding convention is to spell out the complete metric name in the code one way or another.

Reply:
good idea, yeah, the scope is putting metric name into fragments....